src/community/wps-longitudinal-profile/src/main/java/org/geoserver/wps/longitudinal/LongitudinalProfileProcess.java

```diff
@@ -66,18 +70,45 @@ public class LongitudinalProfileProcess implements GeoServerProcess, DisposableB
 
     private static final GeometryFactory GEOMETRY_FACTORY = new GeometryFactory();
     private static final String SEP = System.lineSeparator();
+    // for reference, 18k points would currently use 1MB of memory (56 bytes per point)
+    public static final int DEFAULT_MAX_POINTS = 50000;
+    // default chunk size for parallel processing (used to be 1000, load tests shows this 10% faster)
+    public static final int DEFAULT_CHUNK_SIZE = 5000;
 
     private final GeoServer geoServer;
+    private int chunkSize;
```

Comment:
chunkSize and maxPoints fields are resolved during object construction, we may make them final.

Reply:
I'm now wondering if maybe the opposite is even better: make them input parameters per WPS request if not too complex?

Reply:
Input parameters, nope, no way, they are a way to safeguard the stability of the server, we cannot allow the random process caller to make them too small or too big.